ompi/mca/mtl/ofi/mtl_ofi_component.c

```diff
@@ -82,6 +85,33 @@ ompi_mtl_ofi_component_register(void)
                                     MCA_BASE_VAR_SCOPE_READONLY,
                                     &prov_exclude);
 
+    control_progress = "manual";
+    mca_base_component_var_register(&mca_mtl_ofi_component.super.mtl_version,
+                                    "control_progress",
+                                    "Specify control progress model (default: manual). Set to auto for auto progress.",
+                                    MCA_BASE_VAR_TYPE_STRING, NULL, 0, 0,
+                                    OPAL_INFO_LVL_3,
+                                    MCA_BASE_VAR_SCOPE_READONLY,
+                                    &control_progress);
+
+    data_progress = "auto";
+    mca_base_component_var_register(&mca_mtl_ofi_component.super.mtl_version,
+                                    "data_progress",
+                                    "Specify data progress model (default: auto). Set to manual for manual progress.",
+                                    MCA_BASE_VAR_TYPE_STRING, NULL, 0, 0,
+                                    OPAL_INFO_LVL_3,
+                                    MCA_BASE_VAR_SCOPE_READONLY,
+                                    &data_progress);
+
+    av_type = "map";
+    mca_base_component_var_register(&mca_mtl_ofi_component.super.mtl_version,
+                                    "av_type",
+                                    "Specify AV type to use (default: map). Set to table for FI_AV_TABLE AV type.",
+                                    MCA_BASE_VAR_TYPE_STRING, NULL, 0, 0,
+                                    OPAL_INFO_LVL_3,
+                                    MCA_BASE_VAR_SCOPE_READONLY,
+                                    &av_type);
+
     return OMPI_SUCCESS;
```

Comment:
@hppritcha This is a good place to use the mca_base_var_enum class. You can let the variable system handle the conversion of string->int.